app/jobs/doi_minting_job.rb

```diff
@@ -4,6 +4,7 @@ class DoiMintingJob < ApplicationJob
   queue_as :doi
 
   def perform(resource)
+    sleep(2)
```

Comment:
What was the difference in behavior that you said that you saw when you omitted this `sleep` call? Did you see it in both your development and test environments? Or was it in just one or the other?

Reply:
So test environment is fine either way, but in local dev & the preview branch, without the sleep, v1 work versions do not successfully mint a doi. It will mint one on v2 of the same work successfully though. I had trouble figuring out why this is because it is hitting the DoiMintingJob but not throwing any type of error that I can tell. I only found out the sleep works because I put in a byebug to dig further & it worked that time so I replaced it with a sleep & then everything worked as expected.

Reply:
Though that being said, this is the only spot the sleep made a difference. I also tried one before `if @resource.mint_doi_requested` in `publish_controller` and at the beginning of the call method for MintDoiAsync and neither changed anything

Reply:
I think I was able to get this to work locally without the sleep by disabling ActiveJob in the config for the dev environments:

https://github.com/psu-libraries/scholarsphere/blob/main/config/environments/development.rb#L19-L20

Looks like we override SideKiq in the development environment and just use ActiveJob instead.  The minting still failed for me, but I think it's rejecting my url because it's a localhost url.  But the job queued and ran.  This doesn't explain why it's not working on the preview branch, though.  It should be using SideKiq there.

Reply:
Also, I don't know why it works with SideKiq but not ActiveJob.

Reply:
Ah I see Eric has a fix for this.  Using SideKiq probably just delayed the job enough to mimic what the sleep was doing.  My comments can be ignored.